modified seo tags

imApoorva36 opened the pull request:
# Description

Added the seo tags for the website

Fixes # [(91)](https://github.com/IEEE-NITK/corpus/issues/91)

## Dependencies

List any dependencies that are required for this change.

## Type of Change

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration. (for bug fix / breaking change)
_Put an `x` in the boxes that apply_

- [ ] Test A
- [ ] Test B

---

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

nishant-nayak commented on corpus/templates/base.html:
```diff
@@ -4,19 +4,109 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1.0, user-scalable=no">
+
+    <!-- Google Authorship Markup -->
+    <!-- Social: Twitter -->
+    <meta name="twitter:card" content="summary_large_image">
+    <meta name="twitter:site" content="@{{site.twitter_username}}">
+    <meta name="twitter:title" content="{% if page.title %}{{ page.title }}{% else %}{{ site.title }}{% endif %}">
+    <meta name="twitter:description"
+        content="{% if post.description %}{{ post.description|striptags|linebreaksbr|truncatechars:160 }}{% else %}{{ site_description }}{% endif %}">
+    <meta property="twitter:image:src"
+        content="{% static 'img/blog-image.png' %}">
```
We don't have access to IEEE NITK's Twitter anymore. Delete these tags.

imApoorva36 replied:
Done, also optimized Google+ tags along with Facebook / Open Graph tags...

nishant-nayak commented on corpus/templates/base.html:
In general, there are a lot of templated strings in this file (places where the value starts with `{{` and ends with `}}`). These values were fetched from the Jekyll config in the old site, now we need to hardcode them in.

Take a look at the [_config.yml](https://github.com/IEEE-NITK/ieee-nitk.github.io/blob/master/_config.yml) file in the old repo and copy all the templated values into this.

imApoorva36 replied:
Yes, done

nishant-nayak commented on corpus/templates/base.html:
```diff
@@ -4,19 +4,109 @@
     <link href="https://fonts.googleapis.com/css2?family=Ubuntu&display=swap" rel="stylesheet">
     <link rel="stylesheet"
           href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,400,0,0"/>
+    <link href="https://cdn.jsdelivr.net/npm/daisyui@4.4.6/dist/full.min.css" rel="stylesheet" type="text/css"/>
```
Why are we fetching the full DaisyUI CSS? Isn't the JS Toolchain in place so that we don't need to import this?

imApoorva36 replied:
done

nishant-nayak reviewed (changes requested):
Left a couple of comments. Additionally, I would recommend you look at what each `meta` tag is really doing and what it represents. It is good information to know for SEO.
Maybe you can even add additional tags for updated SEO.

imApoorva36 commented on corpus/templates/base.html:
```diff
@@ -4,19 +4,109 @@
+    <!-- Refers to distribution of the page -->
+    <meta name="distribution" content="global">
+    <!-- Generator or formatter tag -->
+    <meta name="generator" content="Jekyll">
```
Yes

imApoorva36 commented:
I have reverted with optimized content and I've also added all the images as required and added a favicon too

Do let me know in case of any more changes if required :)

imApoorva36 commented:
> I have reverted with optimized content and I've also added all the images as required and added a favicon too
> 
> Do let me know in case of any more changes if required :)

Also I just realised that I also added the favicon, linked from [issue #76](https://github.com/IEEE-NITK/corpus/issues/76) too, as it was as well along with the icons in the old website icons' collection. Just informing.

nishant-nayak commented:
If you can, can you also replace the text **Corpus** on the navbar with the IEEE NITK logo? The requirement is listed in #76 , if you can add that then we can close both issues in this PR

nishant-nayak commented on corpus/templates/components/navbar_large.html:
```diff
@@ -1,6 +1,9 @@
     <div class="navbar-start">
-        <a href="{% url 'index' %}" class="btn btn-ghost normal-case text-xl">Corpus</a>
+        <a href="{% url 'index' %}" class="btn btn-ghost normal-case text-xl">
+            <img src="../static/img/ieee-nitk.png" alt="IEEE NITK Logo" class="dark-logo w-20 h-10">
+            <img src="../static/img/ieee-nitk-white.png" alt="IEEE NITK Logo" class="light-logo w-20 h-10">
```
Remove the `w-20` reference. This doesn't maintain the original aspect ratio, and instead squeezes the image into a 2:1 aspect ratio which is not what we want.

imApoorva36 replied:
fixed

nishant-nayak commented on corpus/templates/components/navbar_small.html:
```diff
@@ -11,7 +11,10 @@
     <div class="navbar-center">
-        <a href="{% url 'index' %}" class="btn btn-ghost normal-case text-xl">Corpus</a>
+        <a href="{% url 'index' %}" class="btn btn-ghost normal-case text-xl">
+            <img src="../static/img/ieee-nitk.png" alt="IEEE NITK Logo" class="dark-logo w-20 h-10">
+            <img src="../static/img/ieee-nitk-white.png" alt="IEEE NITK Logo" class="light-logo w-20 h-10">
```
Remove the `w-20` reference. This doesn't maintain the original aspect ratio, and instead squeezes the image into a 2:1 aspect ratio which is not what we want.

imApoorva36 replied:
fixed

nishant-nayak reviewed (changes requested):
Requested some small UI changes.

nishant-nayak commented:
![image](https://github.com/IEEE-NITK/corpus/assets/66667161/8955855e-f7bf-44fe-92c9-dd2553e9745e)
The logo in the navbar does not render properly on the SIG pages. Look into why this issue is occurring.

imApoorva36 commented:
> ![image](https://private-user-images.githubusercontent.com/66667161/307539830-8955855e-f7bf-44fe-92c9-dd2553e9745e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg3OTcxMDcsIm5iZiI6MTcwODc5NjgwNywicGF0aCI6Ii82NjY2NzE2MS8zMDc1Mzk4MzAtODk1NTg1NWUtZjdiZi00NGZlLTkyYzktZGQyNTUzZTk3NDVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI0VDE3NDY0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1NzljYjY1NzhkMWE1NjczNGJkOTBhYzYwY2FjMGVlNDJlNGRlYjUyZDc1NTA5OTlkYzlkM2EwZjJmYWI0YjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AzyrtytwRdDxhc8tFVINy5JgZ842PATOZ9n4HA9qDgQ) The logo in the navbar does not render properly on the SIG pages. Look into why this issue is occurring.

ah yes, sorry, it was due to relative path in the navbar logo and not absolute path , fixed for both navbar_large and navbar_small

nishant-nayak commented on corpus/templates/components/navbar_large.html:
```diff
@@ -1,6 +1,9 @@
     <div class="navbar-start">
-        <a href="{% url 'index' %}" class="btn btn-ghost normal-case text-xl">Corpus</a>
+        <a href="{% url 'index' %}" class="btn btn-ghost normal-case text-xl">
+            <img src="/static/img/ieee-nitk.png" alt="IEEE NITK Logo" class="dark-logo h-10">
+            <img src="/static/img/ieee-nitk-white.png" alt="IEEE NITK Logo" class="light-logo h-10">
```
```suggestion
            <img src="{% static 'img/ieee-nitk.png' %}" alt="IEEE NITK Logo" class="dark-logo h-10">
            <img src="{% static 'img/ieee-nitk-white.png' %}" alt="IEEE NITK Logo" class="light-logo h-10">
```

We shouldn't be managing static files with direct paths. In case we choose to migrate static files to a different folder, we would have to reconfigure all the static paths again.

We should instead be using Django's static file manager, which configures the static file path in the template itself (see [here](https://docs.djangoproject.com/en/5.0/howto/static-files/#configuring-static-files)).

imApoorva36 replied:
okay

nishant-nayak commented on corpus/templates/components/navbar_small.html:
```diff
@@ -11,7 +11,10 @@
     <div class="navbar-center">
-        <a href="{% url 'index' %}" class="btn btn-ghost normal-case text-xl">Corpus</a>
+        <a href="{% url 'index' %}" class="btn btn-ghost normal-case text-xl">
+            <img src="/static/img/ieee-nitk.png" alt="IEEE NITK Logo" class="dark-logo h-10">
+            <img src="/static/img/ieee-nitk-white.png" alt="IEEE NITK Logo" class="light-logo h-10">
```
```suggestion
            <img src="{% static 'img/ieee-nitk.png' %}" alt="IEEE NITK Logo" class="dark-logo h-10">
            <img src="{% static 'img/ieee-nitk-white.png' %}" alt="IEEE NITK Logo" class="light-logo h-10">
```

imApoorva36 replied:
done
